kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/batch/KCBQThreadPoolExecutor.java

```diff
@@ -92,12 +87,7 @@ public void awaitCurrentTasks() throws InterruptedException, BigQueryConnectExce
       execute(new CountDownRunnable(countDownLatch));
     }
     countDownLatch.await();
-    maybeThrowEncounteredErrors();
-    if (encounteredErrors.size() > 0) {
-      String errorString = createErrorString(encounteredErrors);
-      throw new BigQueryConnectException("Some write threads encountered unrecoverable errors: "
-                                         + errorString + "; See logs for more detail");
-    }
+    maybeThrowEncounteredError();
```

Comment:
I dont think this bubbles up and kills the connector. I think the `awaitCurrentTasks` method is called in a try block and we are not rethrowing exception type `BigQueryConnectException`

Reply:
My bad, any uncaught exception does kill the task.